### PR TITLE
✨ Delete Runs

### PR DESCRIPTION
**Pull request by murgle2**
Allow admins to delete a users run and shift the other runs to fill in the rank gap. Checks if the user has a next best run and inserts that run.

Notes:
- Does not reset stats gained once the run is deleted
- If the next best run replay does not exist on server, the replay will only be accessible locally
- If a user has tied times as their next best run (same total ticks), then the next best run may not be the most recent
- Since tickrate datatype was changed to decimal this will require db sync

### CODE REVIEW

File: server/src/models/run.js [hunk 1]
```diff
@@ -642,6 +642,98 @@ const updateStats = (resultObj, transaction) => {
...
+				for (const result of results) {
+					result.rank -= 1;
+					result.rankXP = xpSystems.getRankXPForRank(result.rank, completions).rankXP;
+					updates.push(result.save({transaction: transact}));
+				}
+				return Promise.all(updates);
+			});
+		}).then(() => {
+			// if we have a nextBestRun, we need to update oldRank
+			if (nextBestRun) {
```
Comment (Hona) [inline server/src/models/run.js:686]: Make this a guard statement

**Review by Hona — CHANGES_REQUESTED** [review]
Looks pretty good!

**Comment by tsa96** [comment]
Closing, as this is way too outdated to ever be useful. I really appreciate the work done here and sorry that this can't be used. It's very frustrating how disorganized the website work was in this time period, pretty sure we knew the backend was going to need rewriting at this point already, and shouldn't have encouraged people work on issues like this.

**Comment by tsa96** [comment]
Also, I will never accept a PR containing a Gitmoji

**Pull request closed without merging**